content/articles/loading-image-with-glide-in-android/index.md

```diff
@@ -0,0 +1,306 @@
+``` kotlin
+binding.buttonScale2.setOnClickListener{
+Glide.with(this)
+    .load(R.drawable.codingtable)
+    .centerCrop()
+    .into(binding.imageView)
+}
+```
+### b) Fit Center
+Is implemented by using the `fitCenter()` method.
+``` kotlin
+binding.buttonScaling1.setOnClickListener {
+Glide.with(this)
+    .load(R.drawable.codingtable)
+    .fitCenter()
+    .into(binding.imageView)
+}
+```
+### c) Circle Crop
+Is implemented by using the `circleCrop()` method. 
+``` kotlin
+binding.buttonCircular.setOnClickListener {
+Glide.with(this)
+    .load(image)
+    .circleCrop()
+    .into(binding.imageView)
+}
+```
+### Placeholders
+Placeholders are Drawable images shown while the request is in progress.
+
+Once the request is completed successfully, the placeholder disappears. If the request fails and an error `Drawable` is not set, the placeholder will continue to be visible.
+
+In the `MainActivity.kt` file, add the following lines of code:
+
+``` kotlin
+binding.buttonPlaceholder.setOnClickListener {
+Glide.with(this)
+    .load(image)
+    .placeholder(R.drawable.placeholder)
+    .into(binding.imageView)
+}
+```
+### Errors Images
+Error Drawable images are shown if the requested URl model is null and no fallback `Drawable` is set. Error `Drawable` is also displayed if the requested resource permanently fails to load.
+In the `MainActivity.kt` file, add the following lines of code:
+
+``` kotlin
+binding.buttonError.setOnClickListener {
+Glide.with(this)
+    .load("https://encrypted-tbn0.gstatic.com/images?q=tbn")
+    .error(R.drawable.error)
+    .into(binding.imageView)
+}
+``` 
+### Image Caching
+Glide checks many layers of cache before starting a new request for an image:
+1. Active resources - Checks if the requested image is visible in another view.
+2. Memory cache - Checks if the image was recently loaded and still in memory.
+3. Data - Was the data in this image obtained from the written disk cache before?
+
+#### Loading only from the cache
+``` kotlin
+binding.buttonCache.setOnClickListener {
+Glide.with(this)
+    .load(resizeImage)
+    .onlyRetrieveFromCache(true)
+    .into(binding.imageView)
+}
+```
+### Disk Cache Strategies
+``` kotlin
+Glide.with(this)
+    .load(imageUrl)
+    .diskCacheStrategy(DiskCacheStrategy.ALL)
+    .into(binding.imageView)
+```
+### Targets
+In Glide, targets  act as mediators between requests and requestors. Targets are responsible for displaying placeholders, loading resources, and determining the appropriate dimensions for each request.
+
+In the `MainActivity.kt` file add the following lines of code:
+
+``` kotlin
+binding.buttonTarget.setOnClickListener {
+Glide.with(this)
+    .asGif()
+    .load(R.drawable.meditation).into(object : SimpleTarget<GifDrawable>(){
+    override fun onResourceReady(
+    resource: GifDrawable,
+    transition: Transition<in GifDrawable>?
+) {
+    resource.start()
+    binding.imageView.setImageDrawable(resource)
+}
+})
+}
+``` 
```

Comment:
This block of code is not well-formatted. Rectify that.

Reply:
Am done with the corrections

Reply:
Let me take a look 👁.👁